tests/anoph/conftest.py

```diff
@@ -153,10 +153,10 @@ def simulate_genes(self, *, contig, contig_size):
                 gene_start = cur_fwd + inter_size
             else:
                 gene_start = cur_rev + inter_size
-            if gene_start >= contig_size:
+            gene_end = gene_start + gene_size
+            if gene_end >= contig_size:
```

Comment:
This change ensures simulated genes will always be within the requested size range, not truncated by the end of the contig. This in turn fixes the intermittent issues arising with simulating exons.